### PR TITLE
Cem 2517

### PR DESCRIPTION
**Pull request by Hunterjdf**

### CODE REVIEW

**Comment by Hunterjdf** [comment]
Hello sir I have been pushing some edge cases since Tuesday. I don't know if you were able to see them before but you should be able to see them now.

**Comment by Hunterjdf** [comment]
image
image
image
image
image
image
image
-Looks to be a different size due to size of prntScrn
- All are same size as to maintain constant size on Mac's component

**Comment by Hunterjdf** [comment]
Changed the if-else monstrosity into a much nicer switch statement.

File: src/Containers/LimitedTimeBanner/LimitedTimeBanner.tsx [hunk 1]
```diff
@@ -13,10 +13,33 @@ export const LimitedTimeBanner: React.FC<LimitedTimeBannerProps> = ({
 }): React.ReactElement => (
     <BannerBox {...props}>
         <Icon />
-        <p>{hoursRemaining} Hours Remaining</p>
+        <p>{alterTime(hoursRemaining)} Remaining</p>
     </BannerBox>
 ) ;
-
+const alterTime = (value:number)  => {
```
Comment (ralph-dev) [inline src/Containers/LimitedTimeBanner/LimitedTimeBanner.tsx:19]: JSDOC

File: src/Containers/LimitedTimeBanner/LimitedTimeBanner.tsx [hunk 2]
```diff
@@ -13,10 +13,33 @@ export const LimitedTimeBanner: React.FC<LimitedTimeBannerProps> = ({
...
+    {
+        case (value == 1):
+            return ('1 Hour');
+        case (value < 1):
+            return ('< 1 Hour');
+        case  (value >= 24 && value < 168) :
+            value = Math.floor(value / 24) % 60;
+            str = (value + ' Day');
+                break;
+        case (value >= 168 && value < 8736):
```
Comment (ralph-dev) [inline src/Containers/LimitedTimeBanner/LimitedTimeBanner.tsx:31]: You should use const for these magic numbers to give them meaning

**Review by ralph-dev — CHANGES_REQUESTED** [review]

File: src/Containers/LimitedTimeBanner/LimitedTimeBanner.stories.tsx [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { LimitedTimeBanner,LimitedTimeBannerProps} from '../../index';
 import { createStoryTitle} from '../../Constants';
 
 export default {
+
```
Comment (ralph-dev) [inline src/Containers/LimitedTimeBanner/LimitedTimeBanner.stories.tsx:7]: Remove extra line

File: src/Containers/LimitedTimeBanner/LimitedTimeBanner.tsx [hunk 4]
```diff
@@ -2,7 +2,10 @@ import React from 'react';
 import styled from 'styled-components';
 import {Clock} from '@styled-icons/bootstrap/Clock';
 import { MainInterface} from '@Utils/BaseStyles';
-
+const hoursInDay = 24;
```
Comment (ralph-dev) [inline src/Containers/LimitedTimeBanner/LimitedTimeBanner.tsx:5]: Constanta should be all caps

Comment (ralph-dev) [reply src/Containers/LimitedTimeBanner/LimitedTimeBanner.tsx:5]: HOURS_IN_DAY

**Review by ralph-dev — CHANGES_REQUESTED** [review]

**Comment by Hunterjdf** [comment]
image
image
image
image
image
- switched out the main function with moment.duration(value,'minutes').humanize();
- Now it works of a base of  minutes with 120 as default 
- Slightly different wording but it works 
- Other code to catch 0 or less or over a year (min + max)
- comments for Props and function added, based of of other components (those with comments)

File: src/Containers/LimitedTimeBanner/LimitedTimeBanner.tsx [hunk 5]
```diff
@@ -2,34 +2,61 @@ import React from 'react';
 import styled from 'styled-components';
 import {Clock} from '@styled-icons/bootstrap/Clock';
 import { MainInterface} from '@Utils/BaseStyles';
+import moment from 'moment';
 
+console.log(moment.now());
```
Comment (ralph-dev) [inline src/Containers/LimitedTimeBanner/LimitedTimeBanner.tsx:7]: Remove log. Use LogPoints by Visual Studio Code

**Review by ralph-dev — APPROVED** [review]
Lgtm but fix comments and have team review once

**Comment by Hunterjdf** [comment]
Log Turned out to be unnecessary. 
Comments style based on Analytics.tsx

File: src/Containers/LimitedTimeBanner/LimitedTimeBanner.tsx [hunk 6]
```diff
@@ -1,35 +1,63 @@
 import React from 'react';
 import styled from 'styled-components';
 import {Clock} from '@styled-icons/bootstrap/Clock';
-import { MainInterface} from '@Utils/BaseStyles';
+import {MainInterface} from '@Utils/BaseStyles';
+import moment from 'moment';
+
+moment.now();
```
Comment (ralph-dev) [inline src/Containers/LimitedTimeBanner/LimitedTimeBanner.tsx:7]: What's this?

File: src/Containers/LimitedTimeBanner/LimitedTimeBanner.tsx [hunk 7]
```diff
@@ -1,35 +1,63 @@
...
+/**
+* Turns an amount of time in minutes to Hours, Days, Months or a Year
+* Also checks if the time is to short (0 or less) or to long (more than a year) 
+* @param {number} value - The time remaining in minutes  
+* @returns {Strings} - The time remaining in a more concise and understandable form  
+*/
+const alterTime = (value:number)  => {
+    if(value <= 0){
+        return('No Time ')
+    }
```
Comment (ralph-dev) [inline src/Containers/LimitedTimeBanner/LimitedTimeBanner.tsx:44]: Or don't show banner :)

**Review by ralph-dev — APPROVED** [review]
Lgtm

**Comment by Hunterjdf** [comment]
Sir I was having a problem with merging my branch back into cheapreats:master. The changes I a making conflict with what has already been committed last week during the meeting. The "resolve conflicts" button appears unviable do to complexity. The other option is a manual merge however I am less than confident in that procedure as I am not sure where in github I can resolve these conflicts.

**Comment by ralph-dev** [comment]
> Sir I was having a problem with merging my branch back into cheapreats:master. The changes I a making conflict with what has already been committed last week during the meeting. The "resolve conflicts" button appears unviable do to complexity. The other option is a manual merge however I am less than confident in that procedure as I am not sure where in github I can resolve these conflicts. 

Branch off master latest, and copy paste the files into the new branch don't fix the conflicts just start again.

**Pull request closed without merging**